### PR TITLE
Privacy 2024 - Implement CCPA link metric

### PR DESCRIPTION
**Pull request by bstandaert-wustl**
---

**Test websites**:

- https://ebay.com
- https://walmart.com

### CODE REVIEW

**Comment by max-ostapenko** [comment]
@bstandaert-wustl please move it into privacy chapter custom metric under `dist/privacy.js` file

**Comment by max-ostapenko** [comment]
Number of the websites that make sense to test for CCPA compliance is quite small compared to the dataset, particularly considering the crawlers run from Virginia, not California.

Could we define the eligible set of sites to optimize it, e.g. using locale, TLD?

**Comment by pmeenan** [comment]
FWIW, The crawl runs from most of the US regions of Google Cloud so there's SOME California testing (~1/7 of the crawl) but it's not deterministic and hard to say how IP geolocation works for any of the regions.

**Comment by bstandaert-wustl** [comment]
@max-ostapenko From https://petsymposium.org/popets/2022/popets-2022-0030.pdf:

> the CCPA applies to all websites doing business in California, regardless of the domicile of their business or the language of their website.

I don't think we have a reasonable way to filter on "doing business in California". Perhaps we could rank sites by monthly traffic and take only the top fraction (since the CCPA has a minimum user threshold).

**Comment by bstandaert-wustl** [comment]
Per the discussion in Slack, @umariqbal recommends testing this on all sites and framing it as “prevalence of enforcement mechanisms” rather than compliance: https://httparchive.slack.com/archives/C023K97SR8U/p1716560204137519

Are there any other changes you recommend making to this before the deadline?

File: dist/ccpa-link.js
Comment (max-ostapenko) [reply dist/ccpa-link.js]: @bstandaert-wustl sorry, was unsubmitted. Can you see it now?

File: dist/ccpa-link.js
Comment (bstandaert-wustl) [reply dist/ccpa-link.js]: The list comes from this paper: https://petsymposium.org/popets/2022/popets-2022-0030.pdf. They ran a test crawl and then manually checked the results, so all of the exclusion phrases caused a false positive on at least one site. I don't think they identify which sites though.

File: dist/ccpa-link.js
Comment (bstandaert-wustl) [reply dist/ccpa-link.js]: I can, thanks. I don't know where this scenario occurs, but the authors of the linked paper claim it does.

**Comment by tunetheweb** [comment]
@max-ostapenko @bstandaert-wustl what's the latest on this? We need to merge this today if we want to include it in the Web Almanac 2024.

**Comment by bstandaert-wustl** [comment]
@tunetheweb I think the remaining questions are minor - if we aren't able to resolve them before your deadline, can you go ahead and merge this?

**Comment by max-ostapenko** [comment]
Yeah, I see no blockers. 👍

**Review by tunetheweb — APPROVED** [review]

**Comment by tunetheweb** [comment]
Got merge conflicts now @max-ostapenko . Can you resolve?

File: dist/privacy.js
Comment (max-ostapenko) [inline dist/privacy.js:382]:
```suggestion
```

@bstandaert-wustl Isn't `Your ad choices` a valid CCPA link?

File: dist/privacy.js
Comment (max-ostapenko) [inline dist/privacy.js:324]:
```suggestion

```
@bstandaert-wustl linting edit

**Review by max-ostapenko — CHANGES_REQUESTED** [review]

File: dist/privacy.js
Comment (max-ostapenko) [reply dist/privacy.js:381]: Seems it's some alternative framework.

**Comment by max-ostapenko** [comment]
@tunetheweb please merge

**Review by max-ostapenko — APPROVED** [review]

File: dist/privacy.js
Comment (bstandaert-wustl) [reply dist/privacy.js:381]: I believe it's usually for https://en.wikipedia.org/wiki/AdChoices

**Pull request merged**
